Week-3/q1.js

```diff
@@ -0,0 +1,27 @@
+function memoize(fn) {
+  const cache = new Map();
+  return function (...args) {
+    const key = args.toString();
+    if (cache.has(key)) {
+      return cache.get(key);
+    }
+    cache.set(key, fn(...args));
+    return cache.get(key);
+  };
+}
+
+function add(a, b) {
+  return a + b;
+}
+
+function time(fn) {
+  console.time();
+  fn();
+  console.timeEnd();
+}
+
+const memoizeAdd = memoize(add);
+
+time(() => console.log(memoizeAdd(38, 30, 100, 200, 5000)));
+time(() => console.log(memoizeAdd(38, 30, 100, 200, 5000)));
```

Comment:
what if the order of parameters are changed? will it work the same way ?

Reply:
We can sort the parameters and use as key for cache

Reply:
right